saleor/order/migrations/tasks/saleor3_18.py

```diff
@@ -0,0 +1,56 @@
+from django.db.models import F, Func, Min, OuterRef, Subquery
+
+from ....celeryconf import app
+from ...models import Order, OrderLine
+
+# celery task for one batch took 0.1s and 10.54MB at its peak
+BATCH_SIZE = 1000
+
+
+@app.task
+def update_order_subtotals(batch_id=None):
+    if batch_id is None:
+        # Get the minimum order number from the database
+        min_order_number = Order.objects.aggregate(Min("number"))["number__min"] or 0
+        batch_id = min_order_number // BATCH_SIZE
+
+    start_order_number = batch_id * BATCH_SIZE
+    end_order_number = start_order_number + BATCH_SIZE
+
+    # Get the orders for the current batch based on order numbers
+    current_batch_order_numbers = Order.objects.filter(
```

Comment:
Do you really need the logic for the batch size, etc? 
Can't we just pass the `start_number` to the task, and then make:
```
    current_batch_order_numbers = Order.objects.filter(
        number__gte=start_number,
    ).values_list("number", flat=True)[:BATCH_SIZE]
```
And later calling `update_order_subtotals` with the last number from the `current_batch_order_numbers.delay(last_number)`
Then your batch is static, and you don't need the whole logic to maintain the batch size: 12-21 lines